### PR TITLE
feat: Tennis Quickstart 

### LINKED ISSUE #401: Tennis quickstart

### PR DESCRIPTION
**Pull request by zepfred**
## Description of the change

This PR adds a new quickstart for the Tennis problem.

[Screencast from 2024-04-22 09-39-40.webm](https://github.com/TimefoldAI/timefold-quickstarts/assets/11961659/3f2f22db-5146-42bc-9671-6c103aef0ab4)

### Development

- [x] The changes have been covered with tests, if necessary.
- [x] You have a green build, with the exception of the flaky tests.
- [x] UI and JS files are fully tested, the user interface works for all modules affected by your changes (e.g., solve and analyze buttons).
- [x] The network calls work for all modules affected by your changes (e.g., solving a problem).
- [x] The console messages are validated for all modules affected by your changes.

### Code Review

- [x] This pull request includes an explanatory title and description.
- [x] The GitHub issue is linked.
- [ ] At least one other engineer has approved the changes.
- [ ] After PR is merged, inform the reporter.

### CODE REVIEW

File: use-cases/tournament-scheduling/src/main/java/org/acme/tournamentschedule/rest/DemoDataGenerator.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    private List<Team> generateTeams() {
+        return List.of(
+                new Team(0, "Micha"),
+                new Team(1, "Angelika"),
+                new Team(2, "Katrin"),
+                new Team(3, "Susi"),
+                new Team(4, "Irene"),
+                new Team(5, "Kristina"),
+                new Team(6, "Tobias"));
```
Comment (triceo) [inline use-cases/tournament-scheduling/src/main/java/org/acme/tournamentschedule/rest/DemoDataGenerator.java]: Maybe we want to use our own names? :-) Maarten, Geoffrey, Lukas, Chris, Fred, Radek, Maciej, ...

**Review by triceo — APPROVED** [review]
LGTM overall.

I think that from the UI, it's not at all clear what the spots are, and what they're for. Maybe we need a better name for the concept?

File: use-cases/tournament-scheduling/pom.xml [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+    <groupId>org.acme</groupId>
+    <artifactId>timefold-solver-quarkus-tournament-scheduling-quickstart</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.release>17</maven.compiler.release>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <version.io.quarkus>3.9.3</version.io.quarkus>
```
Comment (triceo) [inline use-cases/tournament-scheduling/pom.xml]:
```suggestion
        <version.io.quarkus>3.9.4</version.io.quarkus>
```

**Comment by triceo** [comment]
Also, I am about to merge changes to the Native Github workflow. Please rebase on `development`, see the changed workflow file and add this quickstart there.

**Comment by zepfred** [comment]
> LGTM overall.
> 
> I think that from the UI, it's not at all clear what the spots are, and what they're for. Maybe we need a better name for the concept?

What about replacing `Spot` by `Player`?

**Comment by triceo** [comment]
Could it be a `Match`? A `Game`?

**Comment by zepfred** [comment]
> Match

Yes, it can be.

**Pull request merged**